File: data/readable_format/cpp/file1310.cpp
#include <cstddef> // For size_t
#include <limits>  // For numeric_limits
#include <cstdint> // For fixed-width integer types
#include <cassert>
#include <iostream>

class TypeSizes {
public:
    // Assuming 8, 16, 32, 64 bit integers match the desired sizes.
    using OneByteInt = std::int8_t;
    using TwoByteInt = std::int16_t;
    using FourByteInt = std::int32_t;
    using EightByteInt = std::int64_t;

    // Using float for 4 bytes and double for 8 bytes real.
    using FourByteReal = float;
    using EightByteReal = double;

    static bool byteSizesOK() {
        // Check integer sizes
        bool intSizesOK = (sizeof(OneByteInt) * 8 == 8) &&
                          (sizeof(TwoByteInt) * 8 == 16) &&
                          (sizeof(FourByteInt) * 8 == 32) &&
                          (sizeof(EightByteInt) * 8 == 64);

        // Check real sizes
        bool realSizesOK = (sizeof(FourByteReal) * 8 == 32) &&
                           (sizeof(EightByteReal) * 8 == 64);

        // Check that FourByteReal and EightByteReal are different
        bool realTypesDifferent = !std::is_same<FourByteReal, EightByteReal>::value;

        return intSizesOK && realSizesOK && realTypesDifferent;
    }
};

void testByteSizesOK() {
    assert(TypeSizes::byteSizesOK() == true);
    std::cout << "Byte sizes and type requirements are OK." << std::endl;
}

int main() {
    testByteSizesOK();
    return 0;
}

File: data/readable_format/fortran/file1310.f90
module typeSizes
  implicit none
  integer, parameter :: OneByteInt = selected_int_kind(2)
  integer, parameter :: TwoByteInt = selected_int_kind(4)
  integer, parameter :: FourByteInt = selected_int_kind(9)
  integer, parameter :: EightByteInt = selected_int_kind(18)

  integer, parameter :: FourByteReal = selected_real_kind(P =  6, R =  37)
  integer, parameter :: EightByteReal = selected_real_kind(P = 15, R = 307)

contains

  logical function byteSizesOK()
    integer (kind =  OneByteInt)  :: One   = 1
    integer (kind =  TwoByteInt)  :: Two   = 2
    integer (kind = FourByteInt)  :: Four  = 4
    integer (kind = EightByteInt) :: Eight = 8

    byteSizesOK = .true.

    if (bit_size(One) /= 8) byteSizesOK = .false.
    if (bit_size(Two) /= 16) byteSizesOK = .false.
    if (bit_size(Four) /= 32) byteSizesOK = .false.
    if (bit_size(Eight) /= 64) byteSizesOK = .false.
    if (.not. (FourByteReal > 0 .and. EightByteReal > 0 .and. FourByteReal /= EightByteReal)) then
      byteSizesOK = .false.
    endif
  end function byteSizesOK

end module typeSizes

program testTypeSizes
  use typeSizes
  implicit none

  if (byteSizesOK()) then
    print *, "Byte sizes and type requirements are OK."
  else
    print *, "Byte sizes or type requirements are not met."
  end if

end program testTypeSizes